src/main/groovy/io/seqera/wave/service/builder/impl/ContainerBuildServiceImpl.groovy

```diff
@@ -191,7 +191,7 @@ class ContainerBuildServiceImpl implements ContainerBuildService, JobHandler<Bui
             }
             // save docker config for creds
             Path configFile = null
-            if( req.configJson ) {
+            if( req.configJson && !req.formatSingularity()) {
```

Comment:
Better having the same creds save logic both for docker and singularity. 

This also may be useful https://github.com/seqeralabs/wave/blob/7d997c3935ee31006c804e7bb08aedf560ec6ed6/src/main/groovy/io/seqera/wave/service/job/JobHelper.groovy#L36-L36